cmd/controller/disk.go

```diff
@@ -28,12 +28,16 @@ type DiskInfo struct {
 	Serial           string          // Serial is serial no of a disk
 	Vendor           string          // Vendor is vendor of a disk
 	Path             string          // Path is dev path of a disk like /dev/sda
+	FirmwareRevision string          // FirmwareRevision is the firmare revision for a disk
+	LogicalSize      uint32          // LogicalSize is the Logical size of a disk
+	Version          string          // Version is the disk version
 }
 
 // ProbeIdentifiers contains some keys to uniquely identify each disk by probe
 type ProbeIdentifier struct {
-	Uuid           string // Uuid is uid of disk which is generated by udev probe.
-	UdevIdentifier string // UdevIdentifier(syspath) used to identify disk by udevprobe.
+	Uuid            string // Uuid is uid of disk which is generated by udev probe.
+	UdevIdentifier  string // UdevIdentifier(syspath) used to identify disk by udevprobe.
```

Comment:
udev & smart fields are combined in this struct. will it be good if they are separate.

Reply:
We have put it as one struct only since we don't foresee identifiers to have more than one field as udev identifier has sysPath and  smart Identifier has devPath and further if we add some more probes which have more than one field then they can have structs.
But if still it is good to have struct here then we can have !!